Mathlib/Algebra/Order/Module/Algebra.lean

```diff
@@ -0,0 +1,58 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Algebra.Algebra.Basic
+import Mathlib.Algebra.Order.Module.Defs
+
+/-!
+# Ordered algebras
+
+This file proves properties of algebras where both rings are ordered compatibly.
+-/
+
+variable {α β : Type*} [OrderedCommSemiring α]
+
+section OrderedSemiring
+variable (β)
+variable [OrderedSemiring β] [Algebra α β] [SMulPosMono α β] {a : α}
+
+lemma algebraMap_mono : Monotone (algebraMap α β) :=
+  fun a₁ a₂ ha ↦ by
+    simpa only [Algebra.algebraMap_eq_smul_one] using smul_le_smul_of_nonneg_right ha zero_le_one
+
+lemma algebraMap_nonneg (ha : 0 ≤ a) : 0 ≤ algebraMap α β a := by simpa using algebraMap_mono β ha
```

Comment:
Could you please add a `positivity` extension for `algebraMap` using this lemma and `algebraMap_pos` below?

Reply:
I'm not managing to pattern-match on `algebraMap` because of the `FunLike.coe`. Here's my attempt. I won't have more time to spend on making this work.
```
namespace Mathlib.Meta.Positivity
open Lean Meta Qq Function

/-- Extension for `algebraMap`. -/
@[positivity algebraMap _ _ _]
def evalAlgebraMap : PositivityExt where eval {u α} _zα _pα e := do
  let ~q(@FunLike.coe _ _ _ _ (@algebraMap $α $β $instα $instβ $instαβ) $a) := e | throwError "not algebraMap"
  -- let zα' : Q(Zero ℚ) := q(inferInstance)
  -- let pα' : Q(PartialOrder ℚ) := q(inferInstance)
  -- match ← core zα' pα' a with
  -- | .positive pa =>
  --   let _oα ← synthInstanceQ (q(LinearOrderedField $α) : Q(Type u))
  --   pure (.positive (q((Rat.cast_pos (K := $α)).mpr $pa) : Expr))
  -- | .nonnegative pa =>
  --   let _oα ← synthInstanceQ (q(LinearOrderedField $α) : Q(Type u))
  --   pure (.nonnegative (q((Rat.cast_nonneg (K := $α)).mpr $pa) : Expr))
  -- | .nonzero pa =>
  --   let _oα ← synthInstanceQ (q(DivisionRing $α) : Q(Type u))
  --   let _cα ← synthInstanceQ (q(CharZero $α) : Q(Prop))
  --   pure (.nonzero (q((Rat.cast_ne_zero (α := $α)).mpr $pa) : Expr))
  -- | .none => pure .none


example [OrderedSemiring β] [Algebra α β] [SMulPosMono α β] {a : α} (ha : 0 ≤ a) :
    0 ≤ algebraMap α β a := by positivity
```

Reply:
Can you add a `TODO` about it then?

Reply:
A regular match works fine:
```
@[positivity algebraMap _ _ _]
def evalAlgebraMap : PositivityExt where eval {u β'} _zβ _pβ e := do
  let ~q(@algebraMap $α _ $instα $instβ $instαβ $a) := e | throwError "not algebraMap"
  let zα : Q(Zero $α) := q(inferInstance)
  let (pα' : Q(PartialOrder $α)) ← synthInstanceQ _
  sorry
```
though I haven't tried to write the rest.

Reply:
#9961